clouddriver-kubernetes-v2/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/v2/op/handler/KubernetesDaemonSetHandler.java

```diff
@@ -123,27 +124,32 @@ private Status status(V1beta2DaemonSet daemonSet) {
       return result.unstable("Waiting for daemonset spec update to be observed");
     }
 
-    int desiredReplicas = status.getDesiredNumberScheduled();
-    Integer existing = status.getCurrentNumberScheduled();
-    if (existing == null || desiredReplicas > existing) {
+    int desiredReplicas = defaultToZero(status.getDesiredNumberScheduled());
+    int existing = defaultToZero(status.getCurrentNumberScheduled());
+    if (desiredReplicas > existing) {
       return result.unstable("Waiting for all replicas to be scheduled");
     }
 
-    existing = status.getUpdatedNumberScheduled();
-    if (existing != null && desiredReplicas > existing) {
+    existing = defaultToZero(status.getUpdatedNumberScheduled());
+    if (desiredReplicas > existing) {
       return result.unstable("Waiting for all updated replicas to be scheduled");
     }
 
-    existing = status.getNumberAvailable();
-    if (existing == null || desiredReplicas > existing) {
+    existing = defaultToZero(status.getNumberAvailable());
+    if (desiredReplicas > existing) {
       return result.unstable("Waiting for all replicas to be available");
     }
 
-    existing = status.getNumberReady();
-    if (existing == null || desiredReplicas > existing) {
+    existing = defaultToZero(status.getNumberReady());
+    if (desiredReplicas > existing) {
       return result.unstable("Waiting for all replicas to be ready");
     }
 
     return result;
   }
+
+  // Unboxes an Integer, returning 0 if the input is null
+  private int defaultToZero(@Nullable Integer input) {
```

Comment:
This looked a bit cleaner than doing `Optional.ofNullable(status.someField()).orElse(0)` everywhere.  It also has the really minor benefit of not boxing then unboxing the `0` every time we default, but I did this mostly for readability.  The same function is in two classes, but it seemed tiny enough and trivial enough to no be worth trying to pull out to somewhere more general :man_shrugging: